frontend/src/components/dashboard/aliases/CustomAddressGenerationModal.test.tsx

```diff
@@ -0,0 +1,266 @@
+    const [addressArg, setErrorStateArg] = onPick.mock.calls[0];
+    expect(addressArg).toBe("mymask");
+    expect(typeof setErrorStateArg).toBe("function");
+  });
+
+  test("cancel button triggers onClose", async () => {
+    const user = userEvent.setup();
+    const { onClose } = renderModal({ aliasGeneratedState: false });
+
+    const cancel = screen.getByRole("button", { name: "profile-label-cancel" });
+    await user.click(cancel);
+    expect(onClose).toHaveBeenCalledTimes(1);
+  });
+
+  test("shows composed domain hint using profile subdomain and runtime mozmailDomain", () => {
+    renderModal({ aliasGeneratedState: false });
+    expect(screen.getByText("@vanessa.mozmail.com")).toBeInTheDocument();
+  });
+});
+
+describe("CustomAddressGenerationModal - success flow", () => {
+  test("displays newly created mask and handles 'Copy mask' (copyToClipboard=true)", async () => {
+    const user = userEvent.setup();
+
+    const onUpdate = jest.fn();
+    const findAliasDataFromPrefix = jest
+      .fn()
+      .mockImplementation(
+        (_prefix: string) =>
+          ({}) as unknown as import("../../../hooks/api/aliases").AliasData,
+      );
+
+    const { rerender } = render(
+      <CustomAddressGenerationModal
+        isOpen
+        onClose={jest.fn()}
+        onPick={jest.fn()}
+        onUpdate={onUpdate}
+        subdomain="vanessa"
+        aliasGeneratedState={false}
+        findAliasDataFromPrefix={findAliasDataFromPrefix}
+      />,
+    );
+
+    const input = screen.getByPlaceholderText(
+      "modal-custom-alias-picker-form-prefix-placeholder-redesign",
+    );
+    await user.type(input, "mymask");
+
+    rerender(
+      <CustomAddressGenerationModal
+        isOpen
+        onClose={jest.fn()}
+        onPick={jest.fn()}
+        onUpdate={onUpdate}
+        subdomain="vanessa"
+        aliasGeneratedState
+        findAliasDataFromPrefix={findAliasDataFromPrefix}
+      />,
+    );
+
+    expect(screen.getByText("mymask@vanessa.mozmail.com")).toBeInTheDocument();
+
+    const checkbox = screen.getByRole("checkbox", {
+      name: /popover-custom-alias-explainer-promotional-block-checkbox-label/i,
+    });
+    await user.click(checkbox);
+
+    const copyBtn = screen.getByRole("button", { name: /copy-mask/i });
+    await user.click(copyBtn);
+
+    expect(findAliasDataFromPrefix).toHaveBeenCalledWith("mymask");
+    expect(onUpdate).toHaveBeenCalledTimes(1);
+    const [aliasArg, blockPromosArg, copyToClipboardArg] =
+      onUpdate.mock.calls[0];
+    expect(blockPromosArg).toBe(true);
+    expect(copyToClipboardArg).toBe(true);
+    expect(aliasArg).toBeDefined();
+  });
+
+  test("clicking 'Done' calls onUpdate with copyToClipboard undefined", async () => {
+    const user = userEvent.setup();
+
+    const onUpdate = jest.fn();
+    const findAliasDataFromPrefix = jest
+      .fn()
+      .mockReturnValue(
+        {} as unknown as import("../../../hooks/api/aliases").AliasData,
```

Comment:
question (blocking): This module has `import { AliasData } from "../../hooks/api/aliases";` at the top already. Why do we need to re-import it from this separate path here?